app/jobs/notify_subscriber.rb

```diff
@@ -37,17 +44,20 @@ def send_message_to_webhook(webhook_url, submission)
     )
 
     response.code == 200
-  rescue Socket::ResolutionError
-    # If the subscriber web app has been taken fully offline and its DNS removed, this will fire
+  rescue Socket::ResolutionError, Net::OpenTimeout, Errno::ECONNREFUSED
+    # These errors could be a sign that a client web app has been taken offline, so we
+    # want to catch them in order to run `handle_failure`
     false
   end
 
+  # Returns a boolean indicating whether the subscriber has been deleted.
   def delete_subscriber(subscriber)
     deletion_threshold = ENV.fetch("SUBSCRIBER_FAILED_ATTEMPT_DELETION_THRESHOLD", nil)
 
-    return unless deletion_threshold&.to_i&.positive? && deletion_threshold.to_i <= subscriber.failed_attempts
+    return false unless deletion_threshold&.to_i&.positive? && deletion_threshold.to_i <= subscriber.failed_attempts
 
     subscriber.destroy!
```

Comment:
Why doesn't `destroy!` return said bool?

Reply:
Ah, because destroy either returns a `Subscriber` object or raises an exception. Obviously a `Subscriber` is truthy, but for the sake of clarity I'm trying to use booleans `true/false` rather than `object/nil`